doc/api/next_api_changes/behavior/27492-AL.rst

```diff
@@ -0,0 +1,12 @@
+Image path semantics of toolmanager-based tools
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+Previously, MEP22 ("toolmanager-based") Tools would try to load their icon
+(``tool.image``) relative to the current working directory, or, as a fallback,
+from Matplotlib's own image directory.  Because both approaches are problematic
+for third-party tools (the end-user may change the current working directory
+at any time, and third-parties cannot add new icons in Matplotlib's image
+directory), this behavior is deprecated; instead, ``tool.image`` is now
+interpreted relative to the directory containing the source file where the
+``Tool.image`` class attribute is defined.  (Defining ``tool.image`` as an
```

Comment:
> and does one in practice ever want to reuse a tool icon?

Yes, in fact this is needed to make things work in matplotlib itself (otherwise I may have not noticed): for some tools, the base is implemented in matplotlib.backend_bases but they are then subclassed e.g. in matplotlib.backends.backend_qt (e.g. the save figure tool), which is in another directory.  If I want SaveFigureQt to inherit "image" from the base SaveFigure it only makes sense that "image" is defined relative to the file defining the base SaveFigure class (again, consider the extensibility to third-party backends which may want to do the same).

Reply:
That makes sense.